test/common/router/router_test.cc

```diff
@@ -176,6 +176,7 @@ class RouterTestBase : public testing::Test {
 
     Http::TestHeaderMapImpl headers;
     HttpTestUtility::addDefaultHeaders(headers);
+    EXPECT_CALL(callbacks_, activeSpan()).Times(1);
```

Comment:
These type of accessor expectations aren't really useful. Since `callbacks_` isn't a NiceMock, I would just do something like this in the test constructor.
```
EXPECT_CALL(callbacks_, activeSpan()).WillRepeatedly(Return(...))
```

Then, some test should actually make sure the span has `injectContext` called on it with an expectation.

Reply:
Thanks for the review!

Yup, they were added to appease the existing code / [RouterTest constructor](https://github.com/envoyproxy/envoy/blob/master/test/common/router/router_test.cc#L237)
I think the value it returns is already handled [here](https://github.com/envoyproxy/envoy/blob/953ccd6497c29587d29a84a65796d656418ab122/test/mocks/http/mocks.cc#L66)

I'll see what I can do to tidy it up and making it more precisely check that `injectContext` was called.

Reply:
Yes understood, my suggestion is how to still use a strict mock and appease it for this type of case, then you don't need to repeat it in every test which isn't useful.